src/populateExchange.ts

```diff
@@ -167,23 +164,21 @@ export const extractSelectionsFromQuery = ({
   schema,
   query,
 }: MakeFragmentsFromQueryArg) => {
-  let selections: { selections: SelectionSetNode; type: string }[] = [];
-  let fragments: FragmentDefinitionNode[] = [];
+  const selections: { selections: SelectionSetNode; type: string }[] = [];
+  const fragments: FragmentDefinitionNode[] = [];
   const typeInfo = new TypeInfo(schema);
 
   visit(
     query,
     visitWithTypeInfo(typeInfo, {
       Field: node => {
-        if (!node.selectionSet) {
-          return undefined;
+        if (node.selectionSet) {
+          const type = getTypeName(typeInfo);
+          selections.push({ selections: node.selectionSet, type });
```

Comment:
Personally I'd inline this to

```
selections.push({ selections: node.selectionSet, type: getTypeName(typeInfo) });
```

in case we move to `Babel`